snuba/consumers/consumer.py

```diff
@@ -135,22 +135,37 @@ def close(self) -> None:
         latency_sum = 0.0
         max_end_to_end_latency: Optional[float] = None
         end_to_end_latency_sum = 0.0
+        max_sentry_received_latency: Optional[float] = None
+        sentry_received_latency_sum = 0.0
+
+        def update_latency(value: float, sum: float, max: Optional[float]) -> None:
```

Comment:
I think when you pass floats like this to methods in python, mutations to the arguments only apply within the context of the method:

```
➜  ~ python
Python 3.8.12 (default, Nov  8 2021, 15:11:59) 
[Clang 12.0.5 (clang-1205.0.22.11)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> def update_val(value):
...     value += 1
... 
>>> x = 5
>>> update_val(x)
>>> x
5
```

So if you want to continue with this approach I think you'll need something like a `LatencyRecorder` object which holds onto a reference to something whose contents exist outside the scope of the method

Reply:
yup updated